### PR TITLE
feat(51): Extend lock mechanism

### LINKED ISSUE #51: Users should be able to extend their lock
*Description*
This component presents an existing lock in a table view with "Amount MENTO / Amount veMENTO / Expires on / Withdraw / Extend" as columns.
When user clicks on extend button, they will be presented with a calendar in the pop-up where they can choose a date up to two years in the future. Same as in locker compoenent, only Wednesdays should be available as options.
With extend lock button user can add a slope to their existing lock.

Update from Jluy 23: we made a decision to display only one lock in the UI and not use cliff functionality at all. User can only add more slope when extending the lock.

[Component design in Figma](https://www.figma.com/file/QSQsvksBOspBTONiBBFpqa/MENTO-Airgrab?type=design&node-id=7918%3A2723&mode=design&t=qeUGmP5z9igYRCya-1)

 - [ ] **Scenario: Extend Lock**
  - Given there is a table view of user's existing lock
  - When the user clicks on the "Extend" button for a specific lock
  - Then a calendar pop-up should appear, allowing the user to choose a date up to two years in the future
  - And only Wednesdays should be available as options

### PR DESCRIPTION
**Pull request by tplocic20**
### Description

- Connected the extend button to form and allowed user to extend the lock

### Other changes

- Tiny layout tweaks

### Tested

Go to `My voting power` and select a lock to extend

### Related issues

-  Fixes #51

### CODE REVIEW

**Comment by RyRy79261** [comment]
@tplocic20 I'm getting an error with `timestamp` being undefined when clicking the `My Voting Power` button. 

I'm using the shared account
image

**Review by chapati23 — CHANGES_REQUESTED** [review]
Running into the same issue as @RyRy79261 
image


tested on latest deployment https://governance-925rv5h26-mentolabs.vercel.app/ with this wallet 0x70f2Db6D22593a65Dbb848C94eBACDF0344642F7

**Comment by AnnaJ-P11** [comment]
I had the same error, I fixed that on this PR https://github.com/mento-protocol/governance-ui/pull/70/files --> app/components/locks-list/locks-list.component.tsx `const startDate = new Date(lock.lockCreate[0]?.timestamp * 1000);`

**Review by chapati23 — CHANGES_REQUESTED** [review]
Calendar offers impossible relock dates:

image

File: app/components/_shared/button/button.component.tsx [hunk 1]
```diff
@@ -34,7 +34,9 @@ const Wrapper = ({
   disabled,
 }: WrapperProps) => {
   return (
-    <div className={classNames(disabled && styles.disabled)}>
+    <div
+      className={classNames(block && styles.block, disabled && styles.disabled)}
```
Comment (RyRy79261) [inline app/components/_shared/button/button.component.tsx:38]: Can we swap this for tailwind classes 
```suggestion
      className={`${block && "block"} ${disabled && "cursor-not-allowed  pointer-events-none"}`}
```

File: app/components/_shared/button/button.module.scss [hunk 2]
```diff
@@ -187,4 +179,12 @@
 .disabled {
   cursor: not-allowed;
   pointer-events: none;
+}
+
+.block {
```
Comment (RyRy79261) [inline app/components/_shared/button/button.module.scss:184]: Redundant

File: app/components/_shared/button/button.module.scss [hunk 3]
```diff
@@ -187,4 +179,12 @@
...
+  width: 100%;
+
+  .button {
```
Comment (RyRy79261) [inline app/components/_shared/button/button.module.scss:187]: Redundant

**Comment by RyRy79261** [comment]
Closing in favour of #112

**Pull request closed without merging**